datamodel/node.go

```diff
@@ -202,7 +202,7 @@ type NodePrototype interface {
 // volumes of data, detecting and using this feature can result in significant
 // performance savings.
 type NodePrototypeSupportingAmend interface {
-	AmendingBuilder(base Node) NodeBuilder
+	AmendingBuilder(base Node) NodeAmendingBuilder
```

Comment:
I've started to wonder if NodePrototype is the right place to attach this -- and suspect not.

(yes, I realize this is my own old draft interface, and you're shifting it to be slightly more correct in this patch.  heh.)

Larger comment back in the issue over here: https://github.com/ipld/go-ipld-prime/issues/320#issuecomment-1076447907